### ISSUE TITLE
Double-freeing buffer sends an improperly formed OSC message

### ISSUE THREAD

**Issue by jamshark70** [supercollider/supercollider#3933]
Environment
-----------

* SuperCollider version: 3.10 beta (develop @ b06fb4827)
* Operating system and version: irrelevant

Current Behavior
----------------

```
s.boot;
s.dumpOSC(1);  // irrelevant to behavior, only for debugging

b = Buffer.alloc(s, 100, 1);

b.free;
-> Buffer(nil, nil, nil, nil, nil)
[ "/b_free", 0, 0 ]   // OK

b.free;
WARNING: Buffer has already been freed
[ ",", ]
-> Buffer(nil, nil, nil, nil, nil)
FAILURE IN SERVER: , Command not found
```

Expected Behavior
-----------------

At second `free`:

```
b.free;
WARNING: Buffer has already been freed
-> Buffer(nil, nil, nil, nil, nil)
```

No buffer number is available so no OSC message should be sent.

~~I'm 100% certain~~ I thought this is a regression from 3.9 but I might be wrong. I'm pretty sure I worked on this to prevent bugs on double-freeing buffers but it's now different from how I remember it.

**Comment by jamshark70** [comment]
> it's now different from how I remember it.

https://github.com/supercollider/supercollider/pull/2846/files

This was merged before. Why is it gone now? :angry:

**Comment by snappizz** [comment]
in https://github.com/supercollider/supercollider/commit/20144e9eb5188001f328ff77b45fdf0225519428#diff-2985c955cc42697021ce6442774e9ae1, the if was refactored into `freeMsg`, but `listSendMsg` gets called either way. simple oversight, simple fix. no need to break out the mad emojis.

**Comment by jamshark70** [comment]
The user experience was that I encountered a bug that I had, in fact, fixed at some point in the past.

I was also confused because I was sure I worked on `free` in the past, but there was no evidence of it in "blame" for the `free` method itself. It looked like code had been merged in, and then lost somehow (which I now understand is a weakness of "git blame" output -- reverted changes don't appear at all).

Anyway, I have to take some of the responsibility because I was one of the reviewers for Patrick's PR, and I didn't notice the mistake with `free`.
